classes/class-dintero-checkout-confirmation.php

```diff
@@ -151,21 +151,18 @@ public function get_order_from_reference( $merchant_reference ) {
 	 * @return int
 	 */
 	public function get_order_id_from_reference( $merchant_reference ) {
-		$query_args = array(
-			'fields'      => 'ids',
-			'post_type'   => wc_get_order_types(),
-			'post_status' => array_keys( wc_get_order_statuses() ),
-			'meta_key'    => '_dintero_merchant_reference', // phpcs:ignore WordPress.DB.SlowDBQuery -- Slow DB Query is ok here, we need to limit to our meta key.
-			'meta_value'  => $merchant_reference, // phpcs:ignore WordPress.DB.SlowDBQuery -- Slow DB Query is ok here, we need to limit to our meta key.
+		$orders = wc_get_orders(
+			array(
+				'meta_key'   => '_dintero_merchant_reference',
+				'meta_value' => $merchant_reference,
+				'limit'      => 1,
+				'orderby'    => 'date',
+				'order'      => 'DESC',
+			)
 		);
 
-		$order_ids = get_posts( $query_args );
-
-		if ( empty( $order_ids ) ) {
-			return null;
-		}
-
-		return $order_ids[0];
+		$order = reset( $orders );
+		return $order->get_id() ?? 0;
```

Comment:
Same as the other place you make a meta query. Also consider making it a helper function in the functions file instead, as to not have repeat code throughout the plugin